dis_001_significant_earthquakes/contents/src/__init__.py

```diff
@@ -40,182 +45,312 @@
+        # convert max_age to a string in the format (YYYY-MM-DD)
         max_age = max_age.isoformat()
 
-    # 1. delete by age
+    # if the max_age variable exists
     if max_age:
-        r = cartosql.deleteRows(table, "{} < '{}'".format(time_field, max_age))
+        # delete rows from table which are older than the max_age
+        r = cartosql.deleteRows(table, "{} < '{}'".format(time_field, max_age), CARTO_USER, CARTO_KEY)
+        # get the number of rows that were dropped from the table
         num_dropped = r.json()['total_rows']
 
-    # 2. get sorted ids (old->new)
-    r = cartosql.getFields('cartodb_id', table, order='{}'.format(time_field),
-                           f='csv')
+    # get cartodb_ids from carto table sorted by date (new->old)
+    r = cartosql.getFields('cartodb_id', table, order='{} desc'.format(time_field),
+                           f='csv', user=CARTO_USER, key=CARTO_KEY)
+    # turn response into a list of strings of the ids
     ids = r.text.split('\r\n')[1:-1]
 
-    # 3. delete excess
+    # if number of rows is greater than max_rows, delete excess rows
     if len(ids) > max_rows:
-        r = cartosql.deleteRowsByIDs(table, ids[:-max_rows])
+        r = cartosql.deleteRowsByIDs(table, ids[max_rows:], CARTO_USER, CARTO_KEY)
+        # get the number of rows that have been dropped from the table
         num_dropped += r.json()['total_rows']
     if num_dropped:
         logging.info('Dropped {} old rows from {}'.format(num_dropped, table))
-    return num_dropped
 
-###
-## Accessing remote data
-###
+    return(num_dropped)
+
 def genUID(lat, lon, depth, dt):
+    '''Generate unique id using latitude, longitude, depth and date information from retrieved url
+    INPUT   lat: latitude of the earthquake (string)
+            lon: longitude of the earthquake (string)
+            depth: depth of the earthquake (string)
+            dt: date for which we want to generate id (string)
+    '''
     return '{}_{}_{}_{}'.format(lat, lon, depth, dt)
 
-def processData(existing_ids):
+def processData(url,existing_ids):
+    '''
+    Fetch, process and upload new data
+    INPUT   url: url where you can find the download link for the source data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
+    RETURN  num_new: number of rows of new data sent to Carto table (integer)
+    '''
+    # create an empty list to store new data
     new_data = []
+    # create an empty list to store unique ids of new data we will be sending to Carto table
     new_ids = []
-
+    # create a datetime object with today's date
     startTime = datetime.datetime.today()
 
-    # Iterate backwards 1-week at a time
+    # Retrieve and process data by iterating backwards 1-week at a time
+    # if the current datetime is newer than oldest date of data stored in the Carto
     while startTime > MAX_AGE:
+        # set the end date for collecting data as today's date
```

Comment:
this will only be today's date on the first iteration
```suggestion
        # set the end date for collecting data
```